pages/texture/new-texture-modal.vue

```diff
@@ -350,65 +384,95 @@ export default {
 		},
 		parseJSON() {
 			try {
-				const data = JSON.parse(this.formData.importJSON);
-				this.textures = data;
+				let data = JSON.parse(this.importJSON);
+				if (!data || !data.length) data = [emptyTexture()];
+				// validate parsed data
+				const cleaned = data.map((tex) => {
+					tex.key = crypto.randomUUID();
+					tex.name ||= "";
+					tex.tags ||= [];
+					if (!tex.uses || !tex.uses.length) tex.uses = [emptyUse()];
+					tex.uses = tex.uses.map((use) => {
+						use.key = crypto.randomUUID();
+						use.edition ||= "";
+						if (!use.paths || !use.paths.length) use.paths = [emptyPath()];
+						use.paths = use.paths.map((path) => {
+							path.name ||= "";
+							path.versions ||= [];
+							path.mcmeta ||= false;
+							return path;
+						});
+						return use;
+					});
+					return tex;
+				});
+				this.textures = cleaned;
 			} catch (err) {
 				console.error(err);
 				this.$root.showSnackBar(err, "error");
 			}
 		},
+		copyData() {
+			const data = JSON.stringify(this.cleanedData, null, 2);
+			navigator.clipboard.writeText(data);
```

Comment:
Not sure it's compatible with Firefox, check on MDN

Reply:
checked and it's fine, also we're using the same API for gallery sharing so I'd be surprised if there were problems